web/discord_test.go

```diff
@@ -19,17 +20,19 @@ import (
 // The user has to have an active record in the database and either be an admin
 // or a teacher of the requested course.
 func TestDiscordClient(t *testing.T) {
-	conn, err := grpc.Dial(":9090", grpc.WithInsecure(), grpc.WithBlock())
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, ":9090", grpc.WithInsecure(), grpc.WithBlock())
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal("Connection failed, make sure the server is running on :9090")
 	}
 	defer conn.Close()
 
 	client := pb.NewAutograderServiceClient(conn)
 
 	currentUserID := os.Getenv("DISCORD_USER")
 	if currentUserID == "" {
-		t.Fatal("User ID is not set")
+		t.Skip("This test requires a 'DISCORD_USER' environmental variable with a valid ID of a registered user")
```

Comment:
This check should be before the `DialContext` above, since otherwise the test will block if not able to connect to a server or something...

Reply:
Right. Fixed it.